src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationAnalyzer.cs

```diff
@@ -0,0 +1,111 @@
+				await AnalyzeDacOrDacExtensionDeclarationAsync(symbolContext, pxContext), SymbolKind.NamedType);
+		}
+
+		private static async Task AnalyzeDacOrDacExtensionDeclarationAsync(SymbolAnalysisContext symbolContext, PXContext pxContext)
+		{
+			symbolContext.CancellationToken.ThrowIfCancellationRequested();
+
+			if (!(symbolContext.Symbol is INamedTypeSymbol dacOrDacExtSymbol) || !dacOrDacExtSymbol.IsDacOrExtension(pxContext))
+				return;
+			
+			var dacPropertiesDeclarations  = dacOrDacExtSymbol.GetMembers().OfType<IPropertySymbol>();
+
+			List<AttributeData> keyAttributes = new List<AttributeData>();
+			
+			bool flagIsKey = false;
+			bool flagIsKeyIdentity = false;
+			
+			foreach (var property in dacPropertiesDeclarations)
+			{
+				foreach (var attribute in property.GetAttributes())
+				{
+					if (attribute.NamedArguments.Where(a => a.Key.Contains(IsKey) &&
+															 a.Value.Value is bool boolValue &&
+															 boolValue == true)
+												.Any())
+					{
+						CheckAttributeIdentityOrKey(ref flagIsKey,ref flagIsKeyIdentity, attribute, pxContext);
+						keyAttributes.Add(attribute);
+					}
+				}
+			}
+
+			if(flagIsKey && flagIsKeyIdentity)
+			{
+				List<Location> locations = new List<Location>();
+
+				foreach (var attribute in keyAttributes)
+				{
+					locations.Add(await GetAttributeLocationAsync(attribute, symbolContext.CancellationToken).ConfigureAwait(false));
+				}
+
+				foreach (var attribute in keyAttributes)
+				{
+					Location attributeLocation = await GetAttributeLocationAsync(attribute, symbolContext.CancellationToken).ConfigureAwait(false);
+
+					symbolContext.ReportDiagnostic(
+					Diagnostic.Create(
+						Descriptors.PX1055_DacKeyFieldBound, attributeLocation, locations));
+				}	
+			}
+		}
+
+		private static async Task<Location> GetAttributeLocationAsync(AttributeData attribute, CancellationToken cancellationToken)
+		{
+			if (attribute.ApplicationSyntaxReference == null)
+				return null;
+			SyntaxNode attributeSyntaxNode = await attribute.ApplicationSyntaxReference.GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+
+			return attributeSyntaxNode?.GetLocation();
+		}
+
+		private static bool IsDerivedFromIdentityTypes(AttributeData attribute, PXContext pxContext)
+		{
+			AttributeInformation attributeInformation = new AttributeInformation(pxContext);
+
+			INamedTypeSymbol identityAttributeType = pxContext.FieldAttributes.PXDBIdentityAttribute;
+			INamedTypeSymbol longIdentityAttributeType = pxContext.FieldAttributes.PXDBLongIdentityAttribute;
+
+			return attributeInformation.IsAttributeDerivedFromClass(attribute.AttributeClass, identityAttributeType) ||
+				   attributeInformation.IsAttributeDerivedFromClass(attribute.AttributeClass, longIdentityAttributeType);
+		}
+
+		private static void CheckAttributeIdentityOrKey(ref bool flagIsKey,ref bool flagIsKeyIdentity, AttributeData attribute, PXContext pxContext)
```

Comment:
What is the point of using `ref` here? It causes boxing/unboxing, at least, not talking about design issues.
Use Value Tuples to return multiple values from the method.